dasharo-compatibility/sata-detect.robot

```diff
@@ -0,0 +1,87 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+
+# Library    ../lib/QemuMonitor.py    /tmp/qmp-socket
+# Required setup keywords:
+#    Prepare Test Suite - generic setup keyword for all tests
+# Required teardown keywords:
+#    Log Out And Close Connection - generic setup keyword for all tests,
+#    closes all connections to DUT and PiKVM
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+...                     AND
+...                     Skip If    not ${ESP_SCANNING_SUPPORT}    ESP scanning tests not supported
+...                     AND
+...                     Prepare Required Files For Qemu
+...                     AND
+...                     Prepare EFI Partition With System Files
+...                     AND
+...                     Restore Initial DUT Connection Method
+Suite Teardown      Run Keywords
+...                     Clear Out EFI Partition    AND
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+SAT001.001 ESP Scan should contain SATA if it is present
+    [Documentation]    This test aims to verify that SATA shows up in the
+    ...    boot menu
+    Sleep    1
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SAT001.001 not supported
+    Power On
+    Clear Out EFI Partition
```

Comment:
I am not sure, why do we need to clear EFI partition here?

Reply:
@macpijan I was running this on QEMU to make sure it works as intended, and without clearing EFI partition here the boot menu is full of "on non-block root device" os, which causes prevents us from being able to boot into ubuntu later

Reply:
This thread still awaits a response

Reply:
@macpijan here is the log from a platform: https://cloud.3mdeb.com/index.php/s/KS2ATmokiFppcdL